### PR TITLE
Add Surah range page and add continue functionality for Ayah page

### PR DESCRIPTION
**Pull request by osamasayed**
### Summary
This PR adds the ability to view a range of verses and also adds the ability to continue reading the rest of the Surah when viewing a verse’s page.

### Screenshots

| Before | After |
| ------ | ------ |
|Before 1|After 1|
|Before 2|After 2|
|Before 3|After 3|
|After 4|Before 4|
|Before 5|After 5|
|Before 6|After 6|

### CODE REVIEW

**Comment by muhajirdev** [comment]
tested the functionality ✅

File: src/components/QuranReader/ControlButtons/index.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React from 'react';
```
Comment (tifa2UP) [inline src/components/QuranReader/ControlButtons/index.tsx:1]: Not a big fan of this component name.

File: src/components/QuranReader/ControlButtons/index.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  const isVerse = quranReaderDataType === QuranReaderDataType.Verse;
+  // only show the controls for the verse for now.
+  if (!isVerse) {
+    return <></>;
+  }
+  const { chapterId, verseNumber } = initialData.verses[0];
+  const { versesCount } = getChapterDataById(String(chapterId));
+  const chapterNumber = Number(chapterId);
+  const isFirstSurah = chapterNumber === 1;
+  const isLastSurah = chapterNumber === 114;
```
Comment (tifa2UP) [inline src/components/QuranReader/ControlButtons/index.tsx:23]: nit: I'd rather have this be in the chapters utility.

File: src/components/QuranReader/ControlButtons/index.tsx (hunk 1 above)
Comment (osamasayed) [reply src/components/QuranReader/ControlButtons/index.tsx:1]: How about `NavigationButtons`?

**Comment by mmahalwy** [comment]
#### 📝 Changed routes:


| [**/\[chapterId\]/\[verseId\]**](https://quran-com-frontend-next-5ff4szy92-qurancom.vercel.app/[chapterId]/[verseId]) |
|:-:|
|<a href="https://quran-com-frontend-next-5ff4szy92-qurancom.vercel.app/[chapterId]/[verseId]">Screenshot of /[chapterId]/[verseId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/jjIjCPKPN6GafGir0t-HxZvDfI8hWwCb4xXnXXLeeOHgtvM0vbIXUWF9GbVB5abnIJIzaLLfI8pDAe32_G18WA.png">(view full size)</a> |







Commit 8c5dbb4248a533c4a19a0714398235a1a45db77b (https://quran-com-frontend-next-5ff4szy92-qurancom.vercel.app).

File: src/utils/validator.ts [hunk 3]
```diff
@@ -77,3 +89,53 @@ export const isValidPageId = (juzId: string): boolean => {
...
+ *    or '1-two' or 'one-2'.
+ * 3. If the from verse number exceeds the to verse number. e.g. '8-7'.
+ * 4. If either the from verse number of to verse number exceeds the total number of verses
+ *    for the current chapter e.g. for chapter 1: '7-8' or '8-8'.
+ *
+ * @param {String} chapterId
+ * @param {String} range
+ * @returns {Boolean}
+ */
+export const isValidRange = (chapterId: string, range: string): boolean => {
```
Comment (tifa2UP) [inline src/utils/validator.ts:117]: isValidVerseRange

File: src/components/QuranReader/ControlButtons/index.tsx (hunk 1 above)
Comment (tifa2UP) [reply src/components/QuranReader/ControlButtons/index.tsx:1]: EndOfScrollingControls

**Review by tifa2UP — APPROVED** [review]

**Pull request merged**